### PR TITLE
fixes #19479, #10587 notifications now works with two-pane again

### PR DESCRIPTION
**Pull request by ohadlevy**
- notification is now inside the content div, allowing two-pane server
responses to include it.
- refactored all notifications to use notification helpers (notice,
warning and error) instead of direct flash manipulation
- also added the usage of flash now via the helpers, and fixed
it in a couple of places it was used incorrectly (mostly clone rendering
which does not redirect at all).
- fixed consisteny to remove trailing dots from toast messages (as it
seems most strings are without dot suffix).

### CODE REVIEW

**Review by tbrisker — APPROVED** [review]
Looks good, lets see that tests pass.

**Comment by tbrisker** [comment]
@ohadlevy looks like one test is failing, could you take a look?

**Comment by ohadlevy** [comment]
@tbrisker fixed failing test, I actually changed a notice alert to warning, hence the failure. thanks

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -49,7 +49,7 @@ def update
   def destroy
     @user = find_resource(:destroy_users)
     if @user == User.current
-      notice _("You cannot delete this user while logged in as this user.")
+      warning _("You cannot delete this user while logged in as this user")
```
Comment (ohadlevy) [inline app/controllers/users_controller.rb:52]: this was changed from notice to warning level.

File: webpack/assets/javascripts/foreman_toast_notifications.js [hunk 2]
```diff
@@ -13,15 +13,26 @@ export function notify({message, type, link, sticky = isSticky(type)}) {
...
+  store.dispatch(ToastActions.clearToasts());
+}
+
 function importFlashMessagesFromRails() {
-  $('#notifications').data().flash.forEach(([type, message]) => {
+  const notifications = $('#notifications');
+
+  if (notifications.length === 0) {return;}
+
+  notifications.data().flash.forEach(([type, message]) => {
```
Comment (tbrisker) [inline webpack/assets/javascripts/foreman_toast_notifications.js]: This seems to fail when there is no flash data, which occurs in some of the tests.

**Review by tbrisker — CHANGES_REQUESTED** [review]
Integration tests are still failing.

**Comment by johnpmitsch** [comment]
This fixes http://projects.theforeman.org/issues/19500 for me

**Comment by ohadlevy** [comment]
@tbrisker updated.

File: webpack/assets/javascripts/foreman_toast_notifications.js [hunk 3]
```diff
@@ -13,15 +13,28 @@ export function notify({message, type, link, sticky = isSticky(type)}) {
...
+  store.dispatch(ToastActions.clearToasts());
+}
+
 function importFlashMessagesFromRails() {
-  $('#notifications').data().flash.forEach(([type, message]) => {
+  const notifications = $('#notifications');
+
+  if (notifications.length === 0 ||
+    !notifications.data() ||
+    !notifications.data().flash) {return;}
```
Comment (tbrisker) [inline webpack/assets/javascripts/foreman_toast_notifications.js]: you can safely combine these two checks to `!notifications.data('flash')` (also works for looping below)

**Comment by ohadlevy** [comment]
@tbrisker test failure unrelated.

**Comment by ohadlevy** [comment]
If I read the error on http://projects.theforeman.org/issues/19500#change-86746 correctly, it seems that .data was undefined?

**Comment by ohadlevy** [comment]
Oh I see, tomorrow than :)

**Comment by ohadlevy** [comment]
@tbrisker updated.

**Comment by ohadlevy** [comment]
tests failures seem unrelated?

**Review by tbrisker — APPROVED** [review]

**Comment by tbrisker** [comment]
Thanks @ohadlevy !

**Pull request merged**